medusa/subtitle_providers/legendastv.py

```diff
@@ -258,6 +258,12 @@ def search_titles(self, title, season):
                     else:
                         logger.warning('No season detected for title %d (%s)', title_id, name)
 
+                # discard title mismatches on season
```

Comment:
Now you have parts of the mismatches discard logic in `search_titles` and parts in `query`.